maint/local_python/binding_f77.py

```diff
@@ -463,7 +463,7 @@ def dump_function(v, func_type):
         c_arg_list_B.append(v)
         if func_type == "MPI_Datarep_conversion_function":
             # FIXME: check name mangling
-            code_list_common.append("if (%s == (MPI_Datarep_conversion_function *) mpi_conversion_fn_null_) {" % v)
+            code_list_common.append("if (%s == (MPI_Datarep_conversion_function *) (void *) mpi_conversion_fn_null_) {" % v)
```

Comment:
As far as I can tell, the old binding code used the same comparison with out the (void *) cast. So is the warning because of a new compiler? Or is there something else I'm missing?

Reply:
This was reported by @abrooks98, which I believe was using Intel compiler. The failure from his message:
```
 CC       src/binding/c/coll/lib_libmpi_la-alltoallv.lo
src/binding/fortran/mpif_h/fortran_binding.c: In function 'pmpi_register_datarep_':
src/binding/fortran/mpif_h/fortran_binding.c:8158:31: warning: cast between incompatible function types from 'void (*)(void *, MPI_Fint *, MPI_Fint *, void *, MPI_Offset *, void *, MPI_Fint *)' {aka 'void (*)(void *, int *, int *, void *, long long int *, void *, int *)'} to 'int (*)(void *, MPI_Datatype,  int,  void *, MPI_Offset,  void *)' {aka 'int (*)(void *, int,  int,  void *, long long int,  void *)'} [-Wcast-function-type]
 8158 |     if (read_conversion_fn == (MPI_Datarep_conversion_function *) mpi_conversion_fn_null_) {
      |                               ^
src/binding/fortran/mpif_h/fortran_binding.c:8161:32: warning: cast between incompatible function types from 'void (*)(void *, MPI_Fint *, MPI_Fint *, void *, MPI_Offset *, void *, MPI_Fint *)' {aka 'void (*)(void *, int *, int *, void *, long long int *, void *, int *)'} to 'int (*)(void *, MPI_Datatype,  int,  void *, MPI_Offset,  void *)' {aka 'int (*)(void *, int,  int,  void *, long long int,  void *)'} [-Wcast-function-type]
 8161 |     if (write_conversion_fn == (MPI_Datarep_conversion_function *) mpi_conversion_fn_null_) {
      |     
```

@abrooks98 Could you confirm the compiler details? And also confirm that you don't see this warning before the binding revamp?

Reply:
Actually the same warnings are showing up in our nightly Jenkins with `gcc-9`

Reply:
Ok, the actual issue is the function `mpi_conversion_fn_null` was generated as returning `void`, it should return `int`

Reply:
Unfortunately our internal Jenkins tests for this have expired and are unaccessible. I cannot recall if this was occurring with both gcc and icx or just gcc. If necessary I can test again internally without the patch to provide more info.

Reply:
The old binding uses C signature for `mpi_conversion_fn_null_`, that's why there isn't warnings. I'll do the same then.